src/index.js

```diff
@@ -29,70 +29,91 @@ function App() {
     // we only call this retrieving from localStorage once, because we only need to load from saved items when page first loads up, never again
 
     useEffect(() => {
-        localStorage.setItem(LOCAL_STORAGE_KEY_DATA, JSON.stringify(todos))
-    },[todos]); // this effect only fires if the todos array changes
+        localStorage.setItem(LOCAL_STORAGE_KEY_DATA, JSON.stringify(todos));
+    },[todos]); // this effect only fires if the todos array changes: which is toggle complete and remove tasks
 
-    function handleAddItem(e) {
+    const handleAddItem = () => {
         const currentInput =  toDoInput.current.value;
         const pickedDueDate = dueDateInput.current.value;
+        const currentTodos = [...todos];
 
-        if(currentInput && pickedDueDate) {
-            setToDos(prevList => {
-                return [...prevList, { id:uuidv4(), name: currentInput, due: pickedDueDate, completed: false}]
-            })
+        if(currentInput && pickedDueDate){ // will implement validations in next step
+            setToDos([...currentTodos, { id:uuidv4(), name: currentInput, due: pickedDueDate, completed: false}]);
```

Comment:
Continue previous comment. You're not even mutating `todos`, why copy it 😉 
I think its good that you're being careful about not to mutate unnecessary variables. The next step would be thinking about "do I really need to do it, or am I just doing it because people said so".

Also for readability you can create the new todo item object first then add it to the list afterwards. This is so much easier to read. Also if creating large object or array it's good to make each field a new line.
```suggestion
            const newTodoItem = { 
                id: uuidv4(),
                name: currentInput,
                due: pickedDueDate,
                completed: false
            }
            setToDos([...todos, newTodoItem]);
```